bindings/python/example_scripts/export_account_totals.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+# export_account_totals -- export the account totals of all accounts.
+
+# Copyright (C) 2020 Hong Xu <hong@topbug.net>
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# @author Hong Xu <hong@topbug.net>
+
+# Usage:
+#
+#    python3 export_account_totals.py /path/to/gnucash_book_file > output.csv
+
+import copy
+import sys
+
+from gnucash import GncNumeric, Session, SessionOpenMode
+
+
+def get_all_sub_accounts(account, names=[]):
+    "Iterate over all sub accounts of a given account."
+
+    for child in account.get_children_sorted():
+        child_names = names.copy()
+        child_names.append(child.GetName())
+        yield child, '::'.join(child_names)
+        yield from get_all_sub_accounts(child, child_names)
+
+
+def to_string_with_decimal_point_placed(number: GncNumeric):
+    """Convert a GncNumeric to a string with decimal point placed if permissible.
+    Otherwise returns its fractional representation.
+    """
+
+    number = copy.copy(number)
+    if not number.to_decimal(None):
+        return str(number)
+
+    nominator = str(number.num())
+    point_place = str(number.denom()).count('0')  # How many zeros in the denominator?
+    if point_place == 0:
+        return nominator
+
+    if len(nominator) <= point_place:  # prepending zeros if the nominator is too short
+        nominator = '0' * (point_place - len(nominator)) + nominator
+
+    return '.'.join([nominator[:-point_place - 1], nominator[-point_place:]])
```

Comment:
That's a rather naive algorithm. What if the result is 46398756401/179207364?

Reply:
It output its fractional form (line 45)

Reply:
No, it doesn't. There's a 0 in the denominator.

Reply:
What do you mean? I tried the following and it should return its fractional form on line 45:

```
In [1]: import gnucash

In [2]: a=gnucash.GncNumeric(46398756401, 179207364)

In [3]: a
Out[3]: <gnucash.gnucash_core.GncNumeric at 0x7fce3826e470>

In [4]: str(a)
Out[4]: '46398756401/179207364'

In [5]: a.to_decimal(None)
Out[5]: False

In [6]: str(a)
Out[6]: '46398756401/179207364'
```

Reply:
Sorry, you're right. I was looking at the wrong internal function.

To answer your earlier question about a better helper, there is: `xaccPrintAmount`, but it seems not to be exposed in python.

Reply:
Got it, thanks. Looks like we would also have to expose `GNCPrintAmountInfo` before exposing `xaccPrintAmount`. I'm not very familiar with the Python binding exposure development though. Does this script look good for now, before `xaccPrintAmount` is exposed?

Reply:
Yes.